contracts/p1/mixins/Trading.sol

```diff
@@ -108,14 +120,38 @@ abstract contract TradingP1 is Multicall, ComponentP1, ReentrancyGuardUpgradeabl
     // This is reentrancy-safe because we're using the `nonReentrant` modifier on every method of
     // this contract that changes state this function refers to.
     // slither-disable-next-line reentrancy-vulnerabilities-1
-    function tryTrade(TradeRequest memory req) internal nonReentrant {
+    function tryTrade(TradeKind kind, TradeRequest memory req) internal nonReentrant {
         /*  */
         IERC20 sell = req.sell.erc20();
         assert(address(trades[sell]) == address(0));
 
         IERC20Upgradeable(address(sell)).safeApprove(address(broker), 0);
         IERC20Upgradeable(address(sell)).safeApprove(address(broker), req.sellAmount);
-        ITrade trade = broker.openTrade(req);
+
+        // Only allow starting the next auction back-to-back if msgSender is self
+        // Only time this happens is BackingManager.settleTrade() -> BackingManager.rebalance()
+        // TODO is there a better way to do this?
+        if (_msgSender() != address(this)) {
+            // Warning, Assumption: blocktime <= 12s
+            // Require at least 1 block between auctions of the same kind
+            // This gives space for someone to start one of the opposite kinds of auctions
```

Comment:
We can't compare against a value saved during `settleTrade()` because we want to support the usecase where settleTrade() is higher up in the block. `endTime()` is the best thing we have laying around. We don't know the block number the trade will end at ahead of time.